src/rcc.rs

```diff
@@ -286,13 +283,34 @@ fn into_pre_div(div: u8) -> cfgr2::PREDIV_A {
 }
 
 impl CFGR {
-    /// Uses HSE (external oscillator) instead of HSI (internal RC oscillator) as the clock source.
+    /// Enable HSE (external clock) in crystal mode.
+    /// Uses external oscillator instead of HSI (internal RC oscillator) as the clock source.
     /// Will result in a hang if an external oscillator is not connected or it fails to start.
-    pub fn use_hse<F>(mut self, freq: F) -> Self
+    pub fn use_hse_crystal<F>(mut self, freq: F) -> Self
     where
         F: Into<Hertz>,
     {
         self.hse = Some(freq.into().0);
+        self.hse_bypass = false;
+        self
+    }
+
+    /// Enable HSE (external clock) in bypass mode.
+    /// Uses user provided clock instead of HSI (internal RC oscillator) as the clock source.
+    /// Will result in a hang if an external clock source is not connected.
+    pub fn use_hse_bypass<F>(mut self, freq: F) -> Self
```

Comment:
Maybe make it more clear, what the purpose of the bypass mode is?
I had to google to understand, what hse_bypass is for. 

> HSE crystal oscillator bypassed with external clock

is used for the documentation of the bypass flag. 
Maybe:

```suggestion
    /// Enable HSE (external clock) in bypass mode.
    /// Uses user provided clock instead of a crystal oscillator. 
    /// Will result in a hang if an external clock source is not connected.
    pub fn use_hse_bypass<F>(mut self, freq: F) -> Self
```

Which makes it more clear, that with bypass mode enabled, the microcontroller won't drive the XTAL pins anymore, as it would do, if an external crystal would be present and (bypass mode inactive).